languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -185,28 +377,95 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge member -> def
 }
 
+(method_declaration (formal_parameters (_) @param)) @method
```

Comment:
Something feels wrong about all of the parameters stuff with callable things like methods, constructors, lambdas, etc.

In other languages, the parameters don't point directly to the parent scope, but rather there is some indirection through a jump or pop scope. This might not actually be necessary in Java, because of type information, but I'm not entirely sure, and I want to call it out as a thing that I'm not entirely clear on.

Reply:
We could certainly add a scope to point at the params, point the initial scope of the body at it, and then point it at the lexical scope, but I believe that's (at least operationally) equivalent to adding them to the lexical scope. I'm not averse to the intermediate scope, but I don't see what it adds since it binds lexical variables anyway?